diff --git a/android/native.gradle b/android/native.gradle
index db6a2ead..d417c5c1 100644
--- a/android/native.gradle
+++ b/android/native.gradle
@@ -1,5 +1,5 @@
 project.ext.instabug = [
-    version: '12.8.0'
+    version: '12.7.0.5679262-SNAPSHOT'
 ]

 dependencies {
diff --git a/examples/default/android/app/build.gradle b/examples/default/android/app/build.gradle
index e0bd64bf..8c584432 100644
--- a/examples/default/android/app/build.gradle
+++ b/examples/default/android/app/build.gradle
@@ -182,3 +182,28 @@ dependencies {
 }

 apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
+
+def keyPropertiesFile = rootProject.file("key.properties")
+def keyProperties = new Properties()
+keyProperties.load(new FileInputStream(keyPropertiesFile))
+rootProject.allprojects {
+    repositories {
+        maven {
+            credentials {
+                username keyProperties.getProperty("DREAM11_USERNAME")
+                password keyProperties.getProperty("DREAM11_PASSWORD")
+            }
+            url "https://mvn.instabug.com/nexus/repository/dream-11"
+        }
+        maven {
+            url 'https://oss.sonatype.org/content/repositories/snapshots'
+        }
+        maven {
+            url "https://mvn.instabug.com/nexus/repository/instabug-internal/"
+            credentials {
+                username keyProperties.getProperty("INSTABUG_USERNAME")
+                password keyProperties.getProperty("INSTABUG_PASSWORD")
+            }
+        }
+    }
+}
diff --git a/examples/default/ios/Podfile b/examples/default/ios/Podfile
index b47b0b24..7a00e32e 100644
--- a/examples/default/ios/Podfile
+++ b/examples/default/ios/Podfile
@@ -23,6 +23,7 @@ end

 target 'InstabugExample' do
   config = use_native_modules!
+  pod 'Instabug', :podspec => 'https://ios-releases.instabug.com/custom/feature-app-flow/Instabug.podspec'

   # Flags change depending on the env values.
   flags = get_default_flags()
